staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -456,12 +564,120 @@ func (d *draPlugin) PublishResources(_ context.Context, resources resourceslice.
 	return nil
 }
 
-// RegistrationStatus implements [DRAPlugin.RegistrationStatus].
-func (d *draPlugin) RegistrationStatus() *registerapi.RegistrationStatus {
+// RegistrationStatus returns the result of registration, nil if none received yet.
+func (d *Helper) RegistrationStatus() *registerapi.RegistrationStatus {
 	if d.registrar == nil {
 		return nil
 	}
+	// TODO: protect against concurrency issues.
 	return d.registrar.status
 }
 
-func (d *draPlugin) internal() {}
+// serializeGRPC locks a mutex if serialization is enabled.
+// Either way it returns a method that the caller must invoke
+// via defer.
+func (d *Helper) serializeGRPC() func() {
+	if !d.serialize {
+		return func() {}
+	}
+	d.grpcMutex.Lock()
+	return d.grpcMutex.Unlock
+}
+
+// nodePluginImplementation is a thin wrapper around the helper instance.
```

Comment:
Since these are intended to be private, does it make things cleaner to also move them to a separate `go` file?

Reply:
After dropping the public interface for the helper, the public API is spread across the entire file. An interface allows documenting the full public API in a separate file, but it's not good practice. For example, extending that interface is an API break because there might be other implementations of it, unless extra care is taken with adding an unexported method.

If someone wants to learn how to use this package, they should use `go doc`. For someone implementing it I want to have it in one file.